TelCo.ColorCoder/ColorCodingInitiator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+    /// <summary>
+    /// The 25-pair color code, originally known as even-count color code, 
+    /// is a color code used to identify individual conductors in twisted-pair 
+    /// wiring for telecommunications.
+    /// This class provides the color coding and 
+    /// mapping of pair number to color and color to pair number.
+    /// </summary>
+    class InitiateColorCoding
+    {
+        /// <summary>
+        /// Test code for the class
+        /// </summary>
+        /// <param name="args"></param>
+        private static void Main(string[] args)
+        {
+            int pairNumber = 4;
+            ColorPair testPair1 = ColorFromPairNumber.GetColorFromPairNumber(pairNumber);
+            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
+            Debug.Assert(testPair1.majorColor == Color.White);
+            Debug.Assert(testPair1.minorColor == Color.Brown);
+
+            pairNumber = 5;
+            testPair1 = ColorFromPairNumber.GetColorFromPairNumber(pairNumber);
+            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
+            Debug.Assert(testPair1.majorColor == Color.White);
+            Debug.Assert(testPair1.minorColor == Color.SlateGray);
+
+            pairNumber = 23;
+            testPair1 = ColorFromPairNumber.GetColorFromPairNumber(pairNumber);
+            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
+            Debug.Assert(testPair1.majorColor == Color.Violet);
+            Debug.Assert(testPair1.minorColor == Color.Green);
+
+            ColorPair testPair2 = new ColorPair() { majorColor = Color.Yellow, minorColor = Color.Green };
+            pairNumber = PairNumberFromColor.GetPairNumberFromColor(testPair2);
+            Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}\n", testPair2, pairNumber);
+            Debug.Assert(pairNumber == 18);
+
+            testPair2 = new ColorPair() { majorColor = Color.Red, minorColor = Color.Blue };
+            pairNumber = PairNumberFromColor.GetPairNumberFromColor(testPair2);
+            Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}", testPair2, pairNumber);
+            Debug.Assert(pairNumber == 6);
+
+            Console.WriteLine();
+            //PrintAllPairColorCodes.PrintAllColorCodes();
+            Console.WriteLine("-------Testing-------\n");
+            TestingReferenceManual.TestReferenceManual();
+
+            PrintAllPairColorCodes.PrintAllColorCodes(new MethodsToPrintColors.PrintContentOnConsole());
+            MethodsToPrintColors.TestContentPrintedOnConsole testCodeOnConsole = new MethodsToPrintColors.TestContentPrintedOnConsole();
+            PrintAllPairColorCodes.PrintAllColorCodes(testCodeOnConsole);
+
+            Debug.Assert(testCodeOnConsole.CountOfContentPrintedOnConsole == 25);

```

Comment:
Next time, think of a stronger assert = one that is closer to the customer need. Would the customer be satisfied with 'just' 25 lines of output? Sharpening the tests will help you bring all assumptions to the front.